packages/classic/src/kernel.js

```diff
@@ -106,7 +106,25 @@ class Kernel {
 
     return {
       dispatch(action) {
-        self.dispatch(actions.atCursor(self.query(path), action))
+        const element = self.query(path)
+        if (element) {
+          self.dispatch(actions.atCursor(element, action))
+        } else if (
+          element === undefined &&
+          action.type &&
+          action.type.startsWith('.')
+        ) {
+          while (path.length) {
+            const parentPath = path.slice(0, -1)
+            const parentElement = self.query(parentPath)
+            const isIndexed = I.Iterable.isIndexed(parentElement)
+            if (parentElement && !isIndexed) {
```

Comment:
Needs to be `if (data.isElement(parentElement)) {`. Everything you get from the struct will be iterable (i.e. an immutable.js structure; Maps are also Iterables), but not everything will be an element (i.e. will have a kind).